core/consensus/aura/src/lib.rs

```diff
@@ -734,15 +735,26 @@ mod tests {
 			})
 		}
 
-		fn peer(&self, i: usize) -> &Peer<Self::PeerData> {
+		fn add_peer(&mut self, config: &ProtocolConfig) {
+			let client = Arc::new(test_client::new());
+			let verifier = self.make_verifier(client.clone(), config);
+			let (block_import, justification_import, data) = self.make_block_import(client.clone());
+			let specialization = DummySpecialization;
+			let peer = create_peer(client, block_import, justification_import, data, verifier, specialization, config);
+			self.mut_peers(|peers| {
```

Comment:
> Why do you need to `clone` the peer?

```
error[E0507]: cannot move out of captured variable in an `Fn` closure
   --> core/network/src/test/mod.rs:803:15
    |
801 |         let peer = create_peer(client, block_import, justification_import, data, verifier, specialization, config);
    |             ---- captured outer variable
802 |         self.mut_peers(|peers| {
803 |             peers.push(peer)
    |                        ^^^^ cannot move out of captured variable in an `Fn` closure
```

> `self.peers.push(peer);` is much shorter.
The `self.mut_peers` allows composition of testnets, for example the `JustificationTestnet` below doesn't have `self.peers`, yet it owns a `TestNet` and it can access the `peers` of it via `self.mut_peers`.

Here we could probably do `self.peers` since its the `TestNet` implementation itself..

Reply:
Then `mut_peers` should take `FnOnce`.